Robot2018/src/org/usfirst/frc/team199/Robot2018/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,199 @@
+	private final Encoder rightEnc = RobotMap.rightEnc;
+	private final WPI_TalonSRX dtLeftDrive = RobotMap.dtLeftDrive;
+	private final WPI_TalonSRX dtRightDrive = RobotMap.dtRightDrive;
+	private final SpeedControllerGroup dtLeft = RobotMap.dtLeft;
+	private final SpeedControllerGroup dtRight = RobotMap.dtRight;
+	private final DifferentialDrive robotDrive = RobotMap.robotDrive;
+	private final PIDController turnController = RobotMap.turnController;
+	private final PIDController moveController = RobotMap.moveController;
+//	private final PIDController moveLeftController = RobotMap.moveLeftController;
+//	private final PIDController moveRightController = RobotMap.moveRightController;
+	
+	private final AHRS ahrs = RobotMap.ahrs;
+	private final AnalogGyro dtGyro = RobotMap.dtGyro;
+	private final DoubleSolenoid dtGear = RobotMap.dtGear;
+	
+	public void pushGear(boolean forw) {
+		if(forw) {
+			dtGear.set(DoubleSolenoid.Value.kForward);
+		} else {
+			dtGear.set(DoubleSolenoid.Value.kReverse);
+		}
+	}
+	public void stopGear() {
+		dtGear.set(DoubleSolenoid.Value.kOff);
+	}
+	
+	private double pidOut = 0;
+	public void resetAHRS() {
+		ahrs.reset();
+	}
+	
+	public void setLeftMotor(double value) {
+		dtLeft.set(value);
+	}
+	private void stopLeftMotor() {
+		dtLeft.stopMotor();
+	}
+	
+	public void setRightMotor(double value) {
+		dtRight.set(value);
+	}
+	private void stopRightMotor() {
+		dtRight.stopMotor();
+	}
+	
+	public void stopDrive() {
+		stopRightMotor();
+		stopLeftMotor();
+	}
+	
+	public void arcadeDrive(double speed, double turn) {
+		robotDrive.arcadeDrive(speed, turn, SmartDashboard.getBoolean("Square Drive Values", false));
+	}
+	public void tankDrive(double leftSpeed, double rightSpeed) {
+		robotDrive.tankDrive(leftSpeed, rightSpeed, SmartDashboard.getBoolean("Square Drive Values", false));
+	}
+	
+	public double getDtLeft() {
+		return dtLeft.get();
+	}
+	public double getDtRight() {
+		return dtRight.get();
+	}
+	
+	public double getGyro() {
+		return dtGyro.getAngle();
+	}
+	public void resetGyro() {
+		dtGyro.reset();
+	}
+	
+	public void initDefaultCommand() {
+		setDefaultCommand(new TeleopDrive());
+	}
+	
+	public void disableTurnPid() {
+		turnController.disable();
+	}
+	public void enableTurnPid() {
+		turnController.enable();
+	}
+	public void setSetTurn(double set) {
+		turnController.setSetpoint(set);
+	}
+	
+//	public void disableMoveLeftPid() {
+//		moveLeftController.disable();
+//	}
+//	public void enableMoveLeftPid() {
+//		moveLeftController.enable();
+//	}
+//	public void setSetMoveLeft(double set) {
+//		moveLeftController.setSetpoint(set);
+//	}
+//	
+//	public void disableMoveRightPid() {
+//		moveRightController.disable();
+//	}
+//	public void enableMoveRightPid() {
+//		moveRightController.enable();
+//	}
+//	public void setSetMoveRight(double set) {
+//		moveRightController.setSetpoint(set);
+//	}
+	
+	public void setDistanncePerPuls(boolean left, double dist) {
```

Comment:
missing and 'e' in the method name :)

Reply:
Also, only 1 "n" in distance.

Reply:
And why do all the other similar methods have left and right variants, while this one uses a boolean? I recommend either using the left/right variants here or changing all the other methods to take a "side" parameter that is either LEFT or RIGHT for clarity.

Reply:
done in latest commit